core/state_transition.go

```diff
@@ -224,6 +232,18 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	// 5. there is no overflow when calculating intrinsic gas
 	// 6. caller has enough balance to cover asset transfer for **topmost** call
 
+	if st.msg.IsAA() {
+		code := st.evm.StateDB.GetCode(st.to())
+		if code == nil || len(code) < len(aaPrefix) {
+			return nil, ErrInvalidAAPrefix
+		}
+		for i := range aaPrefix {
+			if aaPrefix[i] != code[i] {
```

Comment:
nit: `code[i] != aaPrefix[i]` since it is the code that is being checked.

Reply:
makes sense, will swap

Reply:
done.